### PR TITLE
📖 docs: update comms team release tasks

### PR DESCRIPTION
**Pull request by oscr**
**What this PR does / why we need it**:

- Adds Netlify links related to updating CAPI book, as was provided [here](https://github.com/kubernetes/k8s.io/pull/4939#issuecomment-1473078669). 
- Adds task and clarifications for comms team tasks. 

cc @joekr

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

File: docs/release/release-tasks.md [hunk 1]
```diff
@@ -290,6 +290,8 @@ for provider authors on how to adopt to the new Cluster API version.
 
 1. Update supported versions in versions.md.
    <br>Prior art: [Update supported versions for v1.3](https://github.com/kubernetes-sigs/cluster-api/pull/6850)
+2. Update Cluster API support and guarantees in CONTRIBUTING.md.
+   <br>Prior art: [Support and guarantees for 1.4](https://github.com/kubernetes-sigs/cluster-api/pull/8308)
```
Comment (sbueringer) [inline docs/release/release-tasks.md:294]: Should we drop step 6 of `Cut a release`? (I think it's the same)

Comment (sbueringer) [reply docs/release/release-tasks.md:294]: Also not sure if it fits better here as this task is done in week 1 and not after the release which was the original intention (but no strong opinion, up to you)

**Review by sbueringer — COMMENTED** [review]
Thx for updating the instructions!

File: docs/release/release-tasks.md [hunk 2]
```diff
@@ -335,7 +337,8 @@ The goal of this task to make the book for the current release available under `
 
 Someone with access to Netlify should:
 1. Change production branch in Netlify the current release branch (e.g. `release-1.4`) to make the book available under `https://cluster-api.sigs.k8s.io`.
-2. Re-deploy via the Netlify UI.
+2. Re-deploy via the Netlify [production branch settings](https://app.netlify.com/sites/kubernetes-sigs-cluster-api/settings/deploys#branches-and-deploy-contexts).
```
Comment (CecileRobertMichon) [inline docs/release/release-tasks.md:340]: Not sure this is pointing to the right URL, the "trigger button" is how I've done this in the past on this page https://app.netlify.com/sites/kubernetes-sigs-cluster-api/deploys

Comment (sbueringer) [reply docs/release/release-tasks.md:340]: I think you're right. My intention was to suggest this URL for the step above to change the production branch setting

Comment (sbueringer) [reply docs/release/release-tasks.md:340]: Let's use this URL for the step above and your URL + "Trigger deploy" button for this step

File: docs/release/release-tasks.md (hunk 1 above)
Comment (ykakarap) [reply docs/release/release-tasks.md:294]: Another suggestion would be to reword step 6 in `Cut a release` to also explicitly mention "update support and guarantees" and also add a note that this step should only be done for minor releases.
(similar to how we have a note that the homebrew step (step 5) should only be done for latest stable release and not for other patch releases.)

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

**Review by killianmuldoon — COMMENTED** [review]
/approve

**Pull request merged**